core/templates/pages/feedback-updates-page/feedback-updates-page.component.ts

```diff
@@ -0,0 +1,370 @@
+  currentFeedbackThreadsSortType!: string;
+  threadSummaries: FeedbackThreadSummary[] = [];
+  numberOfUnreadThreads!: number;
+  feedbackThreadActive!: boolean;
+  paginatedThreadsList: FeedbackThreadSummaryBackendDict[][] = [];
+
+  messageSendingInProgress!: boolean;
+  profilePicturePngDataUrl!: string;
+  profilePictureWebpDataUrl!: string;
+  newMessage!: {
+    'text': string | null;
+  };
+
+  loadingFeedbacks!: boolean;
+  explorationTitle!: string;
+  threadStatus!: string;
+  explorationId!: string;
+  threadId!: string;
+  messageSummaries!: FeedbackMessageSummary[];
+  threadSummary!: FeedbackThreadSummary;
+  communityLibraryUrl = (
+    '/' + AppConstants.PAGES_REGISTERED_WITH_FRONTEND.LIBRARY_INDEX.ROUTE);
+
+  loadingIndicatorIsShown: boolean = false;
+  homeImageUrl: string = '';
+  todolistImageUrl: string = '';
+  progressImageUrl: string = '';
+  windowIsNarrow: boolean = false;
+  directiveSubscriptions = new Subscription();
+
+  constructor(
+    private alertsService: AlertsService,
+    private windowDimensionService: WindowDimensionsService,
+    private dateTimeFormatService: DateTimeFormatService,
+    private focusManagerService: FocusManagerService,
+    private i18nLanguageCodeService: I18nLanguageCodeService,
+    private feedbackUpdatesBackendApiService:
+      FeedbackUpdatesBackendApiService,
+    private loaderService: LoaderService,
+    private threadStatusDisplayService: ThreadStatusDisplayService,
+    private urlInterpolationService: UrlInterpolationService,
+    private userService: UserService,
+    private translateService: TranslateService,
+    private pageTitleService: PageTitleService,
+    private urlService: UrlService
+  ) {}
+
+  ngOnInit(): void {
+    this.loaderService.showLoadingScreen('Loading');
+
+    let userInfoPromise = this.userService.getUserInfoAsync();
+    userInfoPromise.then(userInfo => {
+      const username = userInfo.getUsername();
+      if (username) {
+        this.username = username;
+        [this.profilePicturePngDataUrl, this.profilePictureWebpDataUrl] = (
+          this.userService.getProfileImageDataUrl(username));
+      } else {
+        this.profilePictureWebpDataUrl = (
+          this.urlInterpolationService.getStaticImageUrl(
+            AppConstants.DEFAULT_PROFILE_IMAGE_WEBP_PATH));
+        this.profilePicturePngDataUrl = (
+          this.urlInterpolationService.getStaticImageUrl(
+            AppConstants.DEFAULT_PROFILE_IMAGE_PNG_PATH));
+      }
+    });
+
+    this.fetchFeedbackUpdates();
+
+    Promise.all([
+      userInfoPromise,
+    ]).then(() => {
+      setTimeout(() => {
+        this.loaderService.hideLoadingScreen();
+        // So that focus is applied after the loading screen has dissapeared.
+        this.focusManagerService.setFocusWithoutScroll('ourLessonsBtn');
+      }, 0);
+    }).catch(errorResponse => {
+      // This is placed here in order to satisfy Unit tests.
+    });
+
+    this.loadingFeedbacks = false;
```

Comment:
I'm a bit unclear what this represents. Can we use a more self-explanatory name?

Reply:
Also. note the plural of feedback is feedback (not feedbacks.)

Reply:
I think it is self-explanatory. loading the feedback.
changed feedback**s** to feedback